internal/pkg/cli/app_delete_test.go

```diff
@@ -0,0 +1,328 @@
+		{
+			Name: "staging",
+		},
+	}
+	mockApp := &config.Application{
+		Name: "badgoose",
+	}
+	mockResources := []*stack.AppRegionalResources{
+		{
+			Region:   "us-west-2",
+			S3Bucket: "goose-bucket",
+		},
+	}
+	tests := map[string]struct {
+		appName    string
+		setupMocks func(mocks deleteAppMocks)
+
+		wantedError error
+	}{
+		"happy path": {
+			appName: mockAppName,
+			setupMocks: func(mocks deleteAppMocks) {
+				gomock.InOrder(
+					// deleteSvcs
+					mocks.store.EXPECT().ListServices(mockAppName).Return(mockServices, nil),
+					mocks.svcDeleter.EXPECT().Execute().Return(nil),
+
+					// deleteEnvs
+					mocks.store.EXPECT().ListEnvironments(mockAppName).Return(mockEnvs, nil),
+					mocks.envDeleter.EXPECT().Ask().Return(nil),
+					mocks.envDeleter.EXPECT().Execute().Return(nil),
+
+					// emptyS3bucket
+					mocks.store.EXPECT().GetApplication(mockAppName).Return(mockApp, nil),
+					mocks.deployer.EXPECT().GetRegionalAppResources(mockApp).Return(mockResources, nil),
+					mocks.spinner.EXPECT().Start(deleteAppCleanResourcesStartMsg),
+					mocks.bucketEmptier.EXPECT().EmptyBucket(mockResources[0].S3Bucket).Return(nil),
+					mocks.spinner.EXPECT().Stop(log.Ssuccess(deleteAppCleanResourcesStopMsg)),
+
+					// deleteProjectPipline
+					mocks.pipelineDeleter.EXPECT().Run().Return(nil),
+
+					// deleteAppResources
+					mocks.spinner.EXPECT().Start(deleteAppResourcesStartMsg),
+					mocks.deployer.EXPECT().DeleteApp(mockAppName).Return(nil),
+					mocks.spinner.EXPECT().Stop(log.Ssuccess(deleteAppResourcesStopMsg)),
+
+					// deleteAppConfigs
+					mocks.spinner.EXPECT().Start(deleteAppConfigStartMsg),
+					mocks.store.EXPECT().DeleteApplication(mockAppName).Return(nil),
+					mocks.spinner.EXPECT().Stop(log.Ssuccess(deleteAppConfigStopMsg)),
+
+					// deleteWs
+					mocks.spinner.EXPECT().Start(deleteAppWsStartMsg),
+					mocks.ws.EXPECT().DeleteAll().Return(nil),
+					mocks.spinner.EXPECT().Stop(log.Ssuccess(deleteAppWsStopMsg)),
+				)
+			},
+			wantedError: nil,
+		},
+		"when pipeline manifest does not exist": {
+			appName: mockAppName,
+			setupMocks: func(mocks deleteAppMocks) {
+				gomock.InOrder(
+					// deleteSvcs
+					mocks.store.EXPECT().ListServices(mockAppName).Return(mockServices, nil),
+					mocks.svcDeleter.EXPECT().Execute().Return(nil),
+
+					// deleteEnvs
+					mocks.store.EXPECT().ListEnvironments(mockAppName).Return(mockEnvs, nil),
+					mocks.envDeleter.EXPECT().Ask().Return(nil),
+					mocks.envDeleter.EXPECT().Execute().Return(nil),
+
+					// emptyS3bucket
+					mocks.store.EXPECT().GetApplication(mockAppName).Return(mockApp, nil),
+					mocks.deployer.EXPECT().GetRegionalAppResources(mockApp).Return(mockResources, nil),
+					mocks.spinner.EXPECT().Start(deleteAppCleanResourcesStartMsg),
+					mocks.bucketEmptier.EXPECT().EmptyBucket(mockResources[0].S3Bucket).Return(nil),
+					mocks.spinner.EXPECT().Stop(log.Ssuccess(deleteAppCleanResourcesStopMsg)),
+
+					// deleteProjectPipline
```

Comment:
nit: deleteAppPipeline
```suggestion
					// deleteApplicationPipline
```

Reply:
should this be pipeline btw

Reply:
Done!